### DISCUSSION TITLE
Retrieving layout information within tabs

### DISCUSSION THREAD

**Discussion by jdholtz** [kovidgoyal/kitty#7851] [Q&A] [answered]
I'm currently [making a script](https://github.com/jdholtz/i3-restore/blob/21ad83946d59bdb0971a6a2dc055310439b99b80/programs/plugins/kitty.py#L45) to restore Kitty terminals (OS windows, tabs, layouts) upon logins in i3. I am retrieving a Kitty window's layout from `kitty @ ls` and, as linked above, it contains all information to restore OS windows and tabs. However, from what I've seen in the output (and I may be wrong on this), there isn't enough information in the JSON tree to restore the exact layouts of the windows inside a tab (e.g. vsplit, hsplit, etc.). 

If this layout information is possible for Kitty to store/retrieve, could information be added to the JSON tree?

This is similar to #1197, but this only pertains to window layouts inside of a Kitty tab, not the overall window positions and sizes of Kitty. Also, my script does the restoring, so this is a request to add this information to the tree, not for Kitty to do any restoring itself.

**Comment by kovidgoyal** [comment, marked as answer]
Not sure what you are asking for here. vsplit/hsplit is not window state but how the window split was originally created and is specific to the splits layout. Windows can be re-arranged after being created via the rotate or swap layout actions. I suppose one could add some information to @ls from the splits layout in particular. Not something I have time/interest to do myself, but PR is welcome. Relevant code is the as_dict() method in window.py and splits.py

**Reply by jdholtz** [reply]
Thank you, that answered my question. I will look into creating a PR if I can figure out what information to add. Thanks!
